uber/models.py

```diff
@@ -1654,6 +1654,18 @@ class MPointsForCash(MagModel):
     when        = Column(UTCDateTime, default=lambda: datetime.now(UTC))
 
 
+class PromoCode(MagModel):
+    price           = Column(Integer, default=0)
+    uses            = Column(Integer, default=1)
+    code            = Column(UnicodeText)
+    expiration_date = Column(UTCDateTime, default=c.EPOCH)
+
+
+class PromoCodeUsages(MagModel):
+    promo_id    = Column(UUID, ForeignKey('promo_code.id'))
+    attendee_id = Column(UUID, ForeignKey('attendee.id'))
```

Comment:
Uh, wait, even IF you expect multiple codes and multiple users, postgres/sqlalchemy can still do that with a many-to-many relationship.